clustered/server/src/main/java/org/ehcache/clustered/server/EhcachePassiveEntity.java

```diff
@@ -152,14 +148,16 @@ private void invokeRetirementMessages(PassiveReplicationMessage message) throws
         PassiveReplicationMessage.DestroyServerStoreReplicationMessage destroyMessage = (PassiveReplicationMessage.DestroyServerStoreReplicationMessage) message;
         destroyServerStore(destroyMessage.getStoreName());
         break;
+      case CLIENT_ID_TRACK_OP:
+        ehcacheStateService.getClientMessageTracker().remove(message.getClientId());
+        break;
       default:
         throw new IllegalMessageException("Unknown Retirement Message : " + message);
     }
   }
 
   private void untrackHashInvalidationForEventualCache(InvalidationCompleteMessage message) {
-    InvalidationCompleteMessage invalidationCompleteMessage = message;
-    ehcacheStateService.getInvalidationTracker(invalidationCompleteMessage.getCacheId()).getInvalidationMap().computeIfPresent(invalidationCompleteMessage.getKey(), (key, count) -> {
+    ehcacheStateService.getInvalidationTracker(message.getCacheId()).getInvalidationMap().computeIfPresent(message.getKey(), (key, count) -> {
```

Comment:
Slightly unrelated but the `count--` and `count++` a few lines down should be replaced by `count - 1` and `count + 1` as they can't really modify the `count` value.